From 103830e6e1455665f65be92963ab4e25fe8f8b38 Mon Sep 17 00:00:00 2001
From: Mike Kasick <mike@kasick.org>
Date: Thu, 25 Oct 2012 22:08:55 -0400
Subject: [PATCH 2/6] init: Allow device-specific appends to the property
 service permission structs

Extends device_perms.h API to allow property_perms and control_perms to be
appended on a device-specific basis, without needing to redefine the entire
struct contents.

Change-Id: Ib66422c873bd32670633ce1d721fd843c9098fe7
---
 init/device_perms.h     |   10 ++++++++++
 init/property_service.c |    6 ++++++
 2 files changed, 16 insertions(+)

diff --git a/init/device_perms.h b/init/device_perms.h
index 504c0ab..dbd6ca4 100644
--- a/init/device_perms.h
+++ b/init/device_perms.h
@@ -27,3 +27,13 @@ struct {
      {NULL, 0, 0 }
 };
 */
+
+// Alternatively you can append to the existing property_perms and/or
+// control_perms structs with the following:
+/*
+#define CONTROL_PERMS_APPEND \
+    { "rawip_vsnet1",AID_RADIO, AID_RADIO }, \
+    { "rawip_vsnet2",AID_RADIO, AID_RADIO }, \
+    { "rawip_vsnet3",AID_RADIO, AID_RADIO }, \
+    { "rawip_vsnet4",AID_RADIO, AID_RADIO },
+*/
diff --git a/init/property_service.c b/init/property_service.c
index 92ff851..033a6fc 100644
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -125,6 +125,9 @@ struct {
     { "dolby.", AID_SYSTEM,   0 },
 #endif // DOLBY_END
     { "usb_uicc.", AID_SYSTEM,  0 },
+#ifdef PROPERTY_PERMS_APPEND
+PROPERTY_PERMS_APPEND
+#endif
     { NULL, 0, 0 }
 };
 /* Avoid extending this array. Check device_perms.h */
@@ -142,6 +145,9 @@ struct {
 } control_perms[] = {
     { "dumpstate",AID_SHELL, AID_LOG },
     { "ril-daemon",AID_RADIO, AID_RADIO },
+#ifdef CONTROL_PERMS_APPEND
+CONTROL_PERMS_APPEND
+#endif
      {NULL, 0, 0 }
 };
 /* Avoid extending this array. Check device_perms.h */
-- 
1.7.9.5

